### PR TITLE
Boost: Add foundation pages UI

### PR DESCRIPTION
**Pull request by dilirity**
## Proposed changes:

* Adds a basic UI with save functionality for the foundation pages.
* No validation, nothing. Pretty things will be added later. Code is also re-used so that will be fixed later as well.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

- Make sure you can add text to the lonely textarea and save that text.

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	const validatePatterns = ( value: string ) => {
+		const lines = value
+			.split( '\n' )
+			.map( line => line.trim() )
+			.filter( line => line.trim() !== '' );
+
+		// check if it's a valid regex
+		try {
+			lines.forEach( line => new RegExp( line ) );
```
Comment (donnchawp) [inline projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.tsx]: We need to change this to a different check, for a URL, maybe.

**Review by donnchawp — CHANGES_REQUESTED** [review]
Looks great, except the regexp thing. Need to validate a URL, eventually.

File: projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.tsx (hunk 1 above)
Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.tsx]: Removed in 755c4df

Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.tsx]: I added it back but updated it to make sure we have at least one item in the list.

File: projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.module.scss [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		position: absolute;
+		top: 0;
+		left: -10px;
+		z-index: 9;
+
+		.tooltip :global(.icon-tooltip-container .components-popover__content) {
+			width: 150px;
+		}
+	}
+}
```
Comment (haqadn) [inline projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.module.scss]: Is this used?

Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/foundation-pages/meta/meta.module.scss]: Removed.

**Pull request merged**